File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/command/levelledmobs/LevelledMobsCommand.java
package io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs;

import dev.jorel.commandapi.CommandAPICommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.AboutSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.BackupSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.ConfirmSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.EggSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.HelpSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.KillSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.ReloadSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.RoutineSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.SpawnerSubcommand;
import io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand.SummonSubcommand;

public final class LevelledMobsCommand {

    public static CommandAPICommand createInstance() {
        return new CommandAPICommand("levelledmobs")
            .withSubcommands(
                AboutSubcommand.createInstance(),
                BackupSubcommand.createInstance(),
                ConfirmSubcommand.createInstance(),
                EggSubcommand.createInstance(),
                KillSubcommand.createInstance1(),
                KillSubcommand.createInstance2(),
                ReloadSubcommand.createInstance(),
                RoutineSubcommand.createInstance(),
                SpawnerSubcommand.createInstance(),
                SummonSubcommand.createInstance()
            )
            .withSubcommands(HelpSubcommand.createInstances().toArray(new CommandAPICommand[0]))
            .withAliases("lm", "lvlmobs", "leveledmobs")
            .withPermission("levelledmobs.command.levelledmobs")
            .withShortDescription("Manage the LevelledMobs plugin.")
            .withFullDescription("Manage the LevelledMobs plugin, from re-loading the "
                + "configuration to creating a levelled mob spawn egg item with your "
                + "chosen specifications.");
    }

}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/commands/LevelledMobsCommand.kt
package io.github.arcaneplugins.levelledmobs.commands

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.executors.CommandArguments
import dev.jorel.commandapi.executors.CommandExecutor
import io.github.arcaneplugins.levelledmobs.LevelledMobs
import io.github.arcaneplugins.levelledmobs.commands.subcommands.DebugSubcommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.InfoSubcommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.ReloadSubcommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.RulesSubcommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.SpawnerEggCommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.SpawnerSubcommand
import io.github.arcaneplugins.levelledmobs.commands.subcommands.SummonSubcommand
import io.github.arcaneplugins.levelledmobs.util.Utils
import java.util.function.Consumer
import org.bukkit.command.CommandSender


/**
 * This class handles the command execution of '/levelledmobs'.
 *
 * @author lokka30
 * @since 2.4.0
 */
object LevelledMobsCommand {
    fun createInstance(): CommandAPICommand {
        return CommandAPICommand("LevelledMobs")
            .withSubcommands(
                DebugSubcommand.createInstance(),
                InfoSubcommand.createInstance(),
                ReloadSubcommand.createInstance(),
                RulesSubcommand.createInstance(),
                SpawnerEggCommand.createInstance(),
                SpawnerSubcommand.createInstance(),
                SummonSubcommand.createInstance()
            )
            .withAliases("lm", "lvlmobs", "leveledmobs")
            .withPermission("levelledmobs.command.levelledmobs")
            .withShortDescription("Manage the LevelledMobs plugin.")
            .withFullDescription("Manage the LevelledMobs plugin, from re-loading the "
                    + "configuration to creating a levelled mob spawn egg item with your "
                    + "chosen specifications.")
            .executes(CommandExecutor { sender: CommandSender, _: CommandArguments ->
                var mainUsage = LevelledMobs.instance.messagesCfg.getStringList("command.levelledmobs.main-usage")
                mainUsage = Utils.replaceAllInList(mainUsage, "%prefix%", LevelledMobs.instance.configUtils.prefix)
                mainUsage = Utils.replaceAllInList(mainUsage, "%label%", "")
                mainUsage = Utils.colorizeAllInList(mainUsage)
                mainUsage.forEach(Consumer { message: String ->
                    sender.sendMessage(message)
                })
            })
    }
}